src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -41,6 +41,8 @@ public TypeFactory(OutputLibrary library)
             InputEnumType enumType             => _library.ResolveEnum(enumType).WithNullable(inputType.IsNullable),
             // TODO -- this is a temporary solution until we refactored the type replacement to use input types instead of code model schemas
             InputModelType { Namespace: "Azure.Core.Foundations", Name: "Error" } => SystemObjectType.Create(AzureResponseErrorType, AzureResponseErrorType.Namespace!, null).Type,
+            InputModelType { Namespace: "Azure.Core", Name: "RequestContent" } => KnownParameters.RequestContent.Type,
+            InputModelType { Namespace: "Azure.Core", Name: "RequestContentNullable" } => KnownParameters.RequestContentNullable.Type,
```

Comment:
Maybe we could simplify?
```suggestion
            InputModelType { Namespace: "Azure.Core", Name: "RequestContent" or "RequestContentNullable" } => typeof(RequestContent),
```
since the two types here actually are identical

Reply:
I tend to keep the current logic, because `GetType()` is widely used and we don't know whether this simplification will be right in the future.